### PR TITLE
fix: adding spread support

### LINKED ISSUE #9: mapStateToProps-no-store throws trying to read 'name'
Hit enter way too soon accidentally, but seeing the following stack trace when running the recommended rules over our code. Many thanks for creating this package, looking super forward to using it and wanted to file a few issues I'm seeing running it over our codebase.

```shell
TypeError: Cannot read property 'name' of undefined

      at returnNode.type.ObjectExpression.returnNode.properties.reduce (node_modules/eslint-plugin-react-redux/lib/rules/mapStateToProps-no-store.js:23:24)
          at Array.reduce (<anonymous>)
      at checkFunction (node_modules/eslint-plugin-react-redux/lib/rules/mapStateToProps-no-store.js:22:27)
      at VariableDeclaration.node.declarations.forEach (node_modules/eslint-plugin-react-redux/lib/rules/mapStateToProps-no-store.js:37:11)
          at Array.forEach (<anonymous>)
      at VariableDeclaration (node_modules/eslint-plugin-react-redux/lib/rules/mapStateToProps-no-store.js:31:25)a
      at listeners.(anonymous function).forEach.listener (node_modules/eslint/lib/util/safe-emitter.js:47:58)
          at Array.forEach (<anonymous>)
      at Object.emit (node_modules/eslint/lib/util/safe-emitter.js:47:38)
      at NodeEventGenerator.applySelector (node_modules/eslint/lib/util/node-event-generator.js:251:26)
``` 

The code that triggered the issue is similar to the below:

```jsx
export function SampleHeaderComponent({ showDefaultHeader: showHeader, ...rest }) {
  if (showHeader) {
    return <DefaultHeader show />;
  }

  return <FlowHeader {...rest} />;
}

HeaderStyle.defaultProps = defaultProps;
HeaderStyle.propTypes = propTypes;

const mapStateToProps = state => ({
  ...getSomeStateFromASelector(state),
  showDefaultHeader: showDefaultHeader(state),
});

export default compose(
  withRouter,
  connect(mapStateToProps),
)(SampleHeaderComponent);
```

### PR DESCRIPTION
**Pull request by DianaSuvorova**
fixes #9

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,8 +13,8 @@
   "scripts": {
     "lint": "eslint ./",
     "test": "npm run lint && mocha tests --recursive",
-    "test-single": "mocha tests/lib/rules/mapStateToProps-prefer-parameters-names",
-    "debug-test": "mocha --debug-brk --inspect tests/lib/rules/mapStateToProps-prefer-parameters-names",
+    "test-single": "mocha tests/lib/rules/mapStateToProps-no-store",
```
Comment (ljharb) [inline package.json]: it seems strange that "test-single" would hardcode a specific test here; is this meant to be like `npm run test-single path/to/file` instead?

**Comment by DianaSuvorova** [comment]
@ljharb, it is awesome to see you here! Thanks for joining me.

File: package.json (hunk 1 above)
Comment (DianaSuvorova) [reply package.json]: yes.

Comment (ljharb) [reply package.json]: you could change this to just `mocha`, and then it would work that way :-)

Comment (DianaSuvorova) [reply package.json]: uhm, yes let me just remove these

**Pull request merged**